portal-backend/depmap/compound/legacy_utils.py

```diff
@@ -19,7 +20,36 @@
 
 log = logging.getLogger(__name__)
 
-def _get_deduplicated_experiment_compound_mapping(dataset_id: str) -> list[tuple[CompoundExperiment, Compound]]:
+
+# compound_exp_and_dataset is a weird pre-existing hack for finding the compound experiment. Pass it in here as a param to avoid adding too many
+# independent hacks.
+def does_legacy_dataset_exist_with_compound_experiment(
+    dataset_name: str,
+    compound_exp_and_dataset: List[Tuple[CompoundExperiment, DependencyDataset]],
+):
+    dataset = DependencyDataset.get_dataset_by_name(dataset_name)
+
+    # Find the dataset we are looking for in the list of comound experiment and dataset tuples
+    compound_exp_dataset = next(
+        filter(lambda item: item[1] == dataset, compound_exp_and_dataset), None,
```

Comment:
I have a bit of trouble following this line of code because
1. `compound_exp_and_dataset`is a confusing data type (which has nothing to do with your changes)
2. and there are multiple things happening in this line of code with python functions (`next` and `filter`) that I don't see used super often. 

I'm wondering if it would be possible to just break this out into multiple lines and add some more descriptive variable names? For example, I'm a little bit confused about how the variable `compound_exp_dataset` is different from `compound_exp_and_dataset`

Reply:
Yes, I broke it out into a for loop. I think that will help with this confusion. I also added a lot more comments incrementally in this function to explain what it is doing.